### ISSUE TITLE
Is there a reason that pytorch-forecasting saves loss and logging_metrics multiple times?

### ISSUE THREAD

**Issue by dr-upsilon** [sktime/pytorch-forecasting#1700]
- PyTorch-Forecasting version: 1.1.1
- PyTorch version: 2.4.1
- Python version: 3.12.5
- Operating System: windows

`C:\...miniconda3\envs\envpt\Lib\site-packages\lightning\pytorch\utilities\parsing.py:208: Attribute 'logging_metrics' is an instance of `nn.Module` and is already saved during checkpointing. It is recommended to ignore them using `self.save_hyperparameters(ignore=['logging_metrics'])`.
`

This is caused by `self.save_hyperparameters()` in __init__ method of TemporalFusionTransformer, because save_hyperparameters() uses inspect and frame to identify all the hyperparameters, 

What's the reason to keep it or shall we add handling in __init__?

**Comment by fkiraly** [comment]
Maybe a question for @jdb78.

@dr-upsilon, is this happening in a way that is duplicative with `lightning` core?

**Comment by dr-upsilon** [comment]
> Maybe a question for @jdb78.
> 
> @dr-upsilon, is this happening in a way that is duplicative with `lightning` core?

Yes I think so. BaseModel inherits from HyperparametersMixin from lighting.pytorch.core.mixins.hparams_mixin.

**Comment by fkiraly** [comment]
Looked through this multiple times and I cannot come up with a good reason (without hearing @jdb78 original rationale).

I suppose it is something that changed between different versions of `lightning`, so it made sense once but is now duplicative.

Would you like to contribute a PR?

**Comment by fkiraly** [comment]
@dr-upsilon, monitoring and logging have emerged as a top priority item for the API redesign.

I have hence converted this issue in a more general API redesign issue for monitoring and logging.

* your opinions on the "should" state would be much appreciated!
* if you are interested to contribute more deeply to this, consider joining the `sktime` discord and the `pytorch-forecasting` channel, and/or interacting with the higher level issues.
